### PR TITLE
Update documentation for configuring replication

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

We did not update our documentation for configuring replication since we
embedded ZooKeeper.

Modifications:

- Updated the documentation for configuring replication.
- Miscellaneous:
  - Updated the outdated client code sample in TLS configuration.

Result:

Up-to-date docs

### CODE REVIEW

File: site/src/sphinx/setup-configuration.rst [hunk 1]
```diff
@@ -211,97 +211,120 @@ Core properties
...
+    - ``quorumPort`` (integer)
+
+      - the TCP/IP port number which is used by ZooKeeper for reaching consensus
+
+    - ``electionPort`` (integer)
 
-  - the ZooKeeper timeout, in milliseconds. If ``null``, the default value of '1000 milliseconds' (1 second)
+      - the TCP/IP port number which is used by ZooKeeper for leader election
+
+  - It is highly recommended to have more than 3 replicas because the consensus algorithm requires
```
Comment (minwoox) [inline site/src/sphinx/setup-configuration.rst]: nit: How about adding `odd number`?  have more than 3 and odd number replicas ...

Comment (trustin) [reply site/src/sphinx/setup-configuration.rst]: Fixed - added some links about why odd number of ZK servers are preferred over even number of ZK servers.

File: site/src/sphinx/setup-configuration.rst [hunk 2]
```diff
@@ -211,97 +211,126 @@ Core properties
...
-  - the ZooKeeper timeout, in milliseconds. If ``null``, the default value of '1000 milliseconds' (1 second)
+  - It is highly recommended to have more than 3, preferably odd number of, replicas because the consensus
+    algorithm requires more than half of all replicas to agree with each other to function correctly.
+    If you had 2 replicas, losing just one replica would make your cluster stop to function.
+
+    .. note::
+
+       See `here <http://bytecontinnum.com/2016/09/zookeeper-always-configured-odd-number-nodes/>`_ or
+       `here <https://www.quora.com/HBase-Why-we-run-zookeeper-with-odd-number-of-instance>`_ if you are
+       curious why odd number of replicas are preferred over even number of replicas,
```
Comment (minwoox) [inline site/src/sphinx/setup-configuration.rst]: `,` -> `.` ?

Comment (trustin) [reply site/src/sphinx/setup-configuration.rst]: Oops! Fixed :-)

**Pull request merged**